kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiBatchService.scala

```diff
@@ -0,0 +1,129 @@
+    metadataManager.cancelUnscheduledBatch(batchId)
+  }
+
+  def countBatch(
+      batchType: String,
+      batchUser: Option[String],
+      batchState: Option[String] = None,
+      kyuubiInstance: Option[String] = None): Int = {
+    metadataManager.countBatch(
+      batchType,
+      batchUser.orNull,
+      batchState.orNull,
+      kyuubiInstance.orNull)
+  }
+
+  override def start(): Unit = {
+    assert(running.compareAndSet(false, true))
+    val submitTask: Runnable = () => {
+      while (running.get) {
+        metadataManager.pickBatchForSubmitting(kyuubiInstance) match {
+          case None => Thread.sleep(1000)
+          case Some(metadata) =>
+            val batchId = metadata.identifier
+            info(s"$batchId is picked for submission.")
+            val batchSession = sessionManager.createBatchSession(
+              metadata.username,
+              "anonymous",
+              metadata.ipAddress,
+              metadata.requestConf,
+              metadata.engineType,
+              Option(metadata.requestName),
+              metadata.resource,
+              metadata.className,
+              metadata.requestConf,
+              metadata.requestArgs,
+              Some(metadata), // TODO some logic need to fix since it's not from recovery
+              shouldRunAsync = true)
+            val metadataForUpdate = Metadata(
+              identifier = batchId,
+              kyuubiInstance = kyuubiInstance,
+              requestConf = batchSession.optimizedConf,
+              clusterManager = batchSession.batchJobSubmissionOp.builder.clusterManager())
+            metadataManager.updateMetadata(metadataForUpdate, asyncRetryOnError = false)
+            val sessionHandle = sessionManager.openBatchSession(batchSession)
+            var terminated = false
+            while (!terminated) { // block until batch job finished
+              terminated = sessionManager.getBatchSession(sessionHandle).map { batchSession =>
+                val batchOp = batchSession.batchJobSubmissionOp
+                OperationState.isTerminal(batchOp.getStatus.state)
+              }.getOrElse {
+                error(s"Batch Session $batchId is not existed, marked as finished")
+                true
+              }
+              // should we always treat metastore as the single of truth?
+              //
+              // terminated = metadataManager.getBatchSessionMetadata(batchId) match {
+              //   case Some(metadata) =>
+              //     OperationState.isTerminal(OperationState.withName(metadata.state))
+              //   case None =>
+              //     error(s"$batchId is not existed in metastore, assume it is finished")
+              //     true
+              // }
+              if (!terminated) Thread.sleep(1000)
+            }
+            info(s"$batchId is finished.")
+        }
+      }
+    }
+    (0 until batchExecutor.getCorePoolSize).foreach(_ => batchExecutor.submit(submitTask))
```

Comment:
does it mean that, for batch v2, the max concurrent running batch size = batchExecutor.getCorePoolSize.

It seems is a big performance regression.

Reply:
As the offline discussion, set `kyuubi.batch.submitter.threads` default value to 100 same as `kyuubi.backend.server.exec.pool.size` to achieve same performance with existing v1